packages/smooth_app/lib/background/background_task_folksonomy.dart

```diff
@@ -0,0 +1,228 @@
+  ) => BackgroundTaskFolksonomy._(
+    processName: _operationType.processName,
+    uniqueId: uniqueId,
+    stamp: ';folksonomy;$barcode',
+    barcode: barcode,
+  );
+
+  @override
+  (String, AlignmentGeometry)? getFloatingMessage(
+    AppLocalizations appLocalizations,
+  ) => (
+    appLocalizations.background_task_title_folksonomy,
+    AlignmentDirectional.bottomCenter,
+  );
+
+  @override
+  Future<void> execute(LocalDatabase localDatabase) async {
+    final DaoTransientFolksonomy daoTransientFolksonomy =
+        DaoTransientFolksonomy(localDatabase);
+    final DaoFolksonomy daoFolksonomy = DaoFolksonomy(localDatabase);
+
+    while (true) {
+      final FolksonomyOperation? operation = _getPendingOperations(
+        barcode,
+        daoTransientFolksonomy,
+      )?.firstOrNull;
+
+      if (operation == null) {
+        return;
+      }
+
+      if (!operation.isPerformed) {
+        switch (operation.type) {
+          case FolksonomyAction.add:
+            await _serverAdd(operation.tag);
+            break;
+          case FolksonomyAction.edit:
+            await _serverEdit(operation.tag);
+            break;
+          case FolksonomyAction.remove:
+            await _serverDelete(operation.tag);
+            break;
+          default:
+            throw Exception('Invalid operation type: ${operation.type}');
+        }
+
+        final List<FolksonomyOperation>? pendingOperations =
+            _getPendingOperations(barcode, daoTransientFolksonomy);
+        if (pendingOperations == null || pendingOperations.isEmpty) {
+          return;
+        }
+
+        pendingOperations[0].isPerformed = true;
+        await daoTransientFolksonomy.put(barcode, pendingOperations);
+      }
+
+      await serverRefresh(barcode, daoFolksonomy, localDatabase);
+
+      final List<FolksonomyOperation>? pendingOperations =
+          _getPendingOperations(barcode, daoTransientFolksonomy);
+      if (pendingOperations == null ||
+          pendingOperations.isEmpty ||
+          !pendingOperations.first.isPerformed) {
+        return;
+      }
+
+      pendingOperations.removeAt(0);
+
+      if (pendingOperations.isEmpty) {
+        await daoTransientFolksonomy.delete(barcode);
+      } else {
+        await daoTransientFolksonomy.put(barcode, pendingOperations);
+      }
+    }
+  }
+
+  Future<void> _serverAdd(ProductTag tag) async {
+    final String bearerToken = await _getBearerToken();
+
+    // FIXME: The addProduct tag method does not yet have a way to add a comment.
+    final MaybeError<bool> result = await FolksonomyAPIClient.addProductTag(
+      barcode: tag.barcode,
+      key: tag.key,
+      value: tag.value,
+      bearerToken: bearerToken,
+      uriHelper: ProductQuery.uriFolksonomyHelper,
+    );
+    if (result.isError) {
+      throw Exception('Cannot add product tag: ${result.error}');
+    }
+  }
+
+  Future<void> _serverEdit(ProductTag tag) async {
+    final String bearerToken = await _getBearerToken();
+
```

Comment:
```suggestion
  Future<void> _serverEdit(ProductTag tag, String bearerToken) async {
```
Let the `execute` method deal with bearer token.